core/src/main/java/com/linecorp/armeria/server/ServiceRequestContextBuilder.java

```diff
@@ -267,10 +268,14 @@ public ServiceRequestContext build() {
             }
         }
 
+        final EventLoop serviceWorkerGroup =
+            serviceCfg.serviceWorkerGroup() == null ?
```

Comment:
`serviceWorkerGroup()` always returns the worker group.
How about just calling `eventLoop()` and fix the `eventLoop()` method to bring an `EventLoop` from the worker group?